File: 教材/Python-xq/Matrix-master/Python/mult_matrices.py
import matplotlib.pyplot as plt
from time import time

x = []
y = []

for col_row in range(10, 200, 10):

    print(col_row)

    # se crean las matrices por cada iteracion
    matriz1 = [[1 for i in range(col_row)] for n in range(col_row)]  # crear matriz 1
    matriz2 = [[2 for i in range(col_row)] for n in range(col_row)]  # crear matriz 2
    matriz3 = [[0 for i in range(col_row)] for n in range(col_row)]  # crear matriz 3 que es la que tendra el resultado

    # se hace la multiplicacion de matrices 10 veces para sacar un promedio
    prom = 0  # acumulador se inicia en 0
    for l in range(10):
        time_i = time()
        for i in range(col_row):
            for j in range(col_row):
                matriz3[i][j] = 0
                for k in range(col_row):
                    matriz3[i][j] += matriz1[i][k] * matriz2[k][j]
        time_f = time()
        time_e = time_f - time_i
        elementos = col_row * col_row
        t = time_e / ((col_row * elementos) + ((col_row - 1) * elementos))
        prom += t

    prom = prom / 10
    x.append(col_row)
    y.append(prom)

plt.title('Multiplicación de matrices sin Numpy')
plt.xlabel('Tamaño matrices')
plt.ylabel('Tiempo promedio por operación')
print(x)
print(y)

plt.plot(x, y)
plt.show()

File: 教材/Python-xq/Matrix-master/Java/src/mult_matrices/Mult_matrices.java

package mult_matrices;

import java.util.ArrayList;
import java.util.Scanner;


public class Mult_matrices {

    public static void main(String[] args) {
        ArrayList<Integer> x = new ArrayList<>();
        ArrayList<Double> y = new ArrayList<>();
        for (int col_row = 10; col_row <= 200; col_row += 10) {
            
            int[][] matriz1 = new int[col_row][col_row];
            int[][] matriz2 = new int[col_row][col_row];
            int[][] matriz3 = new int[col_row][col_row];

            for(int i = 0; i < col_row; i++){
                for(int j = 0; j < col_row; j++){
                    matriz1[i][j] = 1;
                    matriz2[i][j] = 2;
                }
            }

            double prom = 0;

            for (int l = 0; l < 10; l++) {
                long t_inicio, t_final;
                t_inicio = System.nanoTime();
                for (int i = 0; i < col_row; i++) {
                    for (int j = 0; j < col_row; j++) {
                        for (int k = 0; k < col_row; k++) {
                            matriz3[i][j] += matriz1[i][k]*matriz2[k][j];
                        }
                    } 
                }
                t_final = System.nanoTime();
                int elementos = col_row * col_row;
                double t = (t_final - t_inicio)/((col_row*elementos) + ((col_row - 1)*elementos));
                prom += t;
            }
            prom = prom / 10;
            x.add(col_row);
            y.add(prom);
        }
        
        System.out.println("Tamaño de la matriz = x");
        System.out.println("Tiempo promedio en hacer una operación (nano segundos = 10^-9) = y");
        
        for (int i = 0; i < 20; i++){ 
            System.out.println("x: " + x.get(i)+ " | y: " + y.get(i) + "\n ---------------");
             
        }
        
   
    }
    
}
